content/tutorials/newsfeed-react.textile

```diff
@@ -0,0 +1,450 @@
+---
+title: React Live NewsFeed Tutorial
+section: tutorials
+index: 90
+---
+
+In this tutorial, we will see how to use Ably "Realtime client library":/realtime to build a realtime NewsFeed with React. We'll be making use Ably's public channel. Once a update is posted, we'll publish it to the public channel. We will also be subscribed to the channel so we can see updates as they are added in realtime.
+
+This tutorial will not include any server-side implementation. Hence, updates won't be persisted to any storage.
+
+React is a JavaScript library for building user interfaces. It is component-based and declarative which makes it easy to build complex UI with it. For more information on React, please refer to their official website "https://reactjs.org":https://reactjs.org.
+
+<img src="/images/tutorials/react-commenting/react_logo.png" style="width: 50%" alt="React logo">
+
+<%= partial 'tutorials/_step-1-setup-free-account' %>
+
+h2. Step 2 - Create a React App
+
+We'll start by creating a new React app. To do this, we'll be using "create-react-app":https://github.com/facebook/create-react-app. create-react-app allows you to create a React app without having to worry about build configurations.
+
+We'll call the app @newsfeed-app@:
+
+```[sh]
+npx create-react-app newsfeed-app
+```
+
+The new React app is ready. We can start it by running:
+
+```[sh]
+cd newsfeed-app
+npm start
+```
+
+This builds the static site, runs a web server and opens a browser. If your browser does not open, navigate to "http://localhost:3000":http://localhost:3000.
+
+Let's now install packages we will need in this tutorial that is ably and semantic-ui:
+
+```[sh]
+npm install --save ably semantic-ui-react semantic-ui-css
+```
+
+"See this step in Github":https://github.com/ably/tutorials
+
+h2. Step 3 - Add Ably publish and subscribe
+Firstly, add this import statement to @index.js@ file to enable use of @semantic-ui-css@ :
+
+```[javascript]
+import 'semantic-ui-css/semantic.min.css'
+```
+
+Now, open empty the file @App.css@ and add the following:
+
+```[stylesheet]
+.App-header {
+  margin-top: 10px
+}
+
+.App-button {
+  margin-top: 5px;
+  float: right;
+}
+
+.App-users {
+  margin-top: 10px;
+  text-align: left
+}
+
+.App-update {
+  margin-bottom: 10px;
+  cursor: pointer;
+  text-align: center
+}
+
+.App-cards {
+  overflow-y: scroll;
+  max-height: 500px
+}
+```
+
+Now, empty the file @App.js@ then add the following imports:
```

Comment:
Now, empty the file @App.js@ then add the following imports:
 =>
Now, replace the contents of @App.js@ with the following imports to begin with: